Mist crashes: no such file or directory, rename 'mist.lokidb~' -> 'mist.lokidb'

#### System information
``````
Version: `0.8.9 developer branch`
OS & Version: osx
Node type: `geth `
``````

After some time Mist left open, mist crashes with the following log message:

```

[2017-02-08 11:00:02.569] [INFO] (ui: browser) - Block arrived  0x4d5e03d9be4c7355b62d6b24531b8f3b1e4091c14924e451aa0d96d8d6efc173
[2017-02-08 11:00:11.154] [ERROR] main - UNCAUGHT EXCEPTION { Error: ENOENT: no such file or directory, rename '/Users/alex/Library/Application Support/Mist/mist.lokidb~' -> '/Users/alex/Library/Application Support/Mist/mist.lokidb'
    at Error (native)
  errno: -2,
  code: 'ENOENT',
  syscall: 'rename',
  path: '/Users/alex/Library/Application Support/Mist/mist.lokidb~',
  dest: '/Users/alex/Library/Application Support/Mist/mist.lokidb' }
[2017-02-08 11:00:11.175] [INFO] main - Defer quitting until sockets and node are shut down
[2017-02-08 11:00:11.175] [INFO] Sockets - Destroy all sockets
[2017-02-08 11:00:11.177] [INFO] Sockets/2 - Disconnecting...
[2017-02-08 11:00:11.180] [INFO] Sockets/3 - Disconnecting...
[2017-02-08 11:00:11.181] [INFO] Sockets/4 - Disconnecting...
[2017-02-08 11:00:11.182] [INFO] Sockets/5 - Disconnecting...
[2017-02-08 11:00:11.182] [INFO] Sockets/6 - Disconnecting...
[2017-02-08 11:00:11.183] [INFO] Sockets/7 - Disconnecting...
[2017-02-08 11:00:11.183] [INFO] Sockets/8 - Disconnecting...
[2017-02-08 11:00:11.183] [INFO] Sockets/9 - Disconnecting...
[2017-02-08 11:00:11.184] [INFO] Sockets/10 - Disconnecting...
[2017-02-08 11:00:11.184] [INFO] Sockets/11 - Disconnecting...
[2017-02-08 11:00:11.184] [INFO] Sockets/12 - Disconnecting...
[2017-02-08 11:00:11.184] [INFO] Sockets/13 - Disconnecting...
[2017-02-08 11:00:11.184] [INFO] Sockets/14 - Disconnecting...
[2017-02-08 11:00:11.185] [INFO] Sockets/15 - Disconnecting...
[2017-02-08 11:00:11.185] [INFO] Sockets/16 - Disconnecting...
[2017-02-08 11:00:11.185] [INFO] Sockets/17 - Disconnecting...
[2017-02-08 11:00:11.185] [INFO] Sockets/18 - Disconnecting...
[2017-02-08 11:00:11.186] [INFO] Sockets/19 - Disconnecting...
[2017-02-08 11:00:11.186] [INFO] Sockets/20 - Disconnecting...
[2017-02-08 11:00:11.186] [INFO] Sockets/21 - Disconnecting...
[2017-02-08 11:00:11.186] [INFO] Sockets/22 - Disconnecting...
[2017-02-08 11:00:11.186] [INFO] Sockets/23 - Disconnecting...
[2017-02-08 11:00:11.187] [INFO] Sockets/24 - Disconnecting...
[2017-02-08 11:00:11.187] [INFO] Sockets/25 - Disconnecting...
[2017-02-08 11:00:11.187] [INFO] Sockets/26 - Disconnecting...
[2017-02-08 11:00:11.187] [INFO] Sockets/27 - Disconnecting...
[2017-02-08 11:00:11.188] [INFO] Sockets/28 - Disconnecting...
[2017-02-08 11:00:11.189] [INFO] Sockets/node-ipc - Disconnecting...
/Users/alex/dev/mist/node_modules/solc/soljson.js:1
(function (exports, require, module, __filename, __dirname, process, global) { var Module;if(!Module)Module=(typeof Module!=="undefined"?Module:null)||{};var moduleOverrides={};for(var key in Module){if(Module.hasOwnProperty(key)){moduleOverrides[key]=Module[key]}}var ENVIRONMENT_IS_WEB=typeof window==="object";var ENVIRONMENT_IS_WORKER=typeof importScripts==="function";var ENVIRONMENT_IS_NODE=typeof process==="object"&&typeof require==="function"&&!ENVIRONMENT_IS_WEB&&!ENVIRONMENT_IS_WORKER;var ENVIRONMENT_IS_SHELL=!ENVIRONMENT_IS_WEB&&!ENVIRONMENT_IS_NODE&&!ENVIRONMENT_IS_WORKER;if(ENVIRONMENT_IS_NODE){if(!Module["print"])Module["print"]=function print(x){process["stdout"].write(x+"\n")};if(!Module["printErr"])Module["printErr"]=function printErr(x){process["stderr"].write(x+"\n")};var nodeFS=require("fs");var nodePath=require("path");Module["read"]=function read(filename,binary){filename=nodePath["normalize"](filename);var ret=nodeFS["readFileSync"](fil

Error: ENOENT: no such file or directory, rename '/Users/alex/Library/Application Support/Mist/mist.lokidb~' -> '/Users/alex/Library/Application Support/Mist/mist.lokidb'
    at Error (native)
```

--- a/package.json
+++ b/package.json
@@ -24,7 +24,7 @@
     "i18next": "^6.1.1",
     "log-rotate": "^0.2.7",
     "log4js": "^1.1.0",
-    "lokijs": "^1.4.2",
+    "lokijs": "^1.4.3",
     "minimongo-standalone": "^1.1.0-3",
     "numeral": "^2.0.4",
     "oboe": "^2.1.3",

